cpp/cmake/thirdparty/get_spdlog.cmake

```diff
@@ -16,21 +16,8 @@
 function(find_and_configure_spdlog)
 
   include(${rapids-cmake-dir}/cpm/spdlog.cmake)
-  rapids_cpm_spdlog(FMT_OPTION "EXTERNAL_FMT_HO" INSTALL_EXPORT_SET cudf-exports)
-  rapids_export_package(BUILD spdlog cudf-exports)
+  rapids_cpm_spdlog(FMT_OPTION "EXTERNAL_FMT_HO")
```

Comment:
I've paired removing the `spdlog` export here with adding this over in `cpp/CMakeLists.txt`:

```cmake
target_linked_libraries(cudf PRIVATE spdlog::spdlog_header_only)
```

Without that, the pip devcontainer build here fails like this while building the `pylibcudf` wheel:

```text
CMake Error at /home/coder/.local/share/venvs/rapids/lib/python3.10/site-packages/cmake/data/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package):
    By not providing "Findspdlog.cmake" in CMAKE_MODULE_PATH this project has
    asked CMake to find a package configuration file provided by "spdlog", but
    CMake did not find one.
    
    Could not find a package configuration file provided by "spdlog" with any
    of the following names:
    
        spdlogConfig.cmake
        spdlog-config.cmake
    
    Add the installation prefix of "spdlog" to CMAKE_PREFIX_PATH or set
    "spdlog_DIR" to a directory containing one of the above files.  If "spdlog"
    provides a separate development package or SDK, be sure it has been
    installed.
```

I think because in devcontainers, `libcudf.so` is built outside of wheel builds, which means that this condition is hit:

https://github.com/rapidsai/cudf/blob/a0c6fc8300bb713721c355feec21e43c83268b47/python/libcudf/CMakeLists.txt#L29-L31

And then later code that makes `spdlog` available isn't run.

Reply:
This seems plausible to me but I'd like a confirmation from a CMake expert. @vyasr or @KyleFromNVIDIA?

Reply:
If libcudf is only using spdlog privately, and its public headers don't depend on spdlog, then I think this is fine.

Reply:
Thank you! I'd thought it was only using it privately, but now that I search around again I'm really not sure.

This makes me think `libcudf`'s logger is considered part of its public API:

https://github.com/rapidsai/cudf/blob/267692490ba245404bf09c526bd61375ba72493b/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md?plain=1#L1084-L1087

Here's the one non-test header directly `#include`-ing `spdlog`:

https://github.com/rapidsai/cudf/blob/267692490ba245404bf09c526bd61375ba72493b/cpp/include/cudf/utilities/logger.hpp#L21

And there I can see that `cudf::logger` is a `spdlog::logger`.

https://github.com/rapidsai/cudf/blob/267692490ba245404bf09c526bd61375ba72493b/cpp/include/cudf/utilities/logger.hpp#L46

So maybe it does need to be exported? But if so then I don't understand how the `pylibcudf` and `cudf` builds and tests could be succeeding with it marked private like this.

One other helpful piece of context that might help explain why this is even coming up now... we were previously patching `spdlog` in `rapids-cmake`, which took it down the "always download with CPM and patch" code paths there. Now that we're not patching, different `rapids-cmake` codepaths are being followed. Something related to exporting must be different between those cases.

Reply:
Since we're using the header-only version of spdlog, if the downstream targets already have include directories that contain spdlog, they don't need to link against the spdlog target to get those headers. So perhaps it is currently just working by accident.

Reply:
Gah great point. Ok let me fiddle with this in devcontainers and try to put the export back.

Reply:
I just pushed https://github.com/rapidsai/cudf/pull/16806/commits/058b0d3baf63b75b95064746edbb8162ad517898 marking `spdlog::spdlog_header_only` as an `INTERFACE` dependency of `cudf::cudf` and adding it to the `cudf-exports` install and build export sets.

Saw that work for me locally in a unified devcontainer. Assuming it works in CI here too, could you take another look and let me know if you think it looks ok?

Reply:
Yes, that sounds right to me. Apologies for not catching this earlier.

Reply:
Didn't see this earlier. Yes, the spdlog logger is currently part of the public API for cudf via the logger. That is why we also wound up with symbol issues around spdlog before we went through and hid all the necessary symbols.

The current code looks close, but I don't think `INTERFACE` is right. It should be a `PUBLIC` dependency because it is required both by consumers of libcudf (because of the spdlog bits in the logger's public interface) and by libcudf internals (to compiled logger.cpp). I suspect that right now `INTERFACE` is working because we transitively inherit the spdlog `INTERFACE` include from rmm.

Reply:
Ah I see, ok that makes sense! Let me test switching it to `PUBLIC`, shouldn't take too long.